src/app/panels/terminal.js

```diff
@@ -663,17 +668,27 @@ class Terminal extends Plugin {
     }
     return self.commands[name]
   }
-  _shell (script, scopedCommands, done) { // default shell
+  async _shell (script, scopedCommands, done) { // default shell
     if (script.indexOf('remix:') === 0) {
       return done(null, 'This type of command has been deprecated and is not functionning anymore. Please run remix.help() to list available commands.')
     }
     var self = this
-    var context = domTerminalFeatures(self, scopedCommands, self.blockchain)
+    if (script.indexOf('remix.') === 0) {
+      // we keep the old feature. This will basically only be called when the command is querying the "remix" object.
+      // for all the other case, we use the Code Executor plugin
+      var context = domTerminalFeatures(self, scopedCommands, self.blockchain)
+      try {
+        var cmds = vm.createContext(Object.assign(self._jsSandboxContext, context, self._jsSandboxRegistered))
+        var result = vm.runInContext(script, cmds)
+        self._jsSandboxContext = Object.assign(cmds, context)
+        return done(null, result)
```

Comment:
Avoid callback structure when using a Promise structure (async/await here) :
Then, you won't need to try catch, as the error will go in the `catch` of the promise.

```javascript
async _shell (script, scopedCommands) {
  ...
  if (script.split('.').shift() === 'remix') {
    ...
    return result
  } else {
    return this.call('remixCodeExecutor', 'execute', script)
  }
}
```

Then I can do : 
```javascript
this._shell(...)
  .then()
  .catch() // <- message error will be handle here.
```
Or 
```javascript
try {
  this._shell(...)
} catch (err) {
  // Handle error here or later.
}
```